threeML/plugins/OGIP/pha.py

```diff
@@ -485,3 +481,355 @@ def __repr__(self):
+
+
+
+            else:
+
+                # There is no ancillary file, so we need to flag it.
+                # There is again an assumption that this is constant
+                # across spectra.
+
+                self._has_ancillary_file = False
+
+            if pha_info['rsp'].rsp_filename is not None:
+
+                self._respfile[key].append(pha_info['rsp'].rsp_filename)
+
+                if len(pha_info['rsp'].rsp_filename) > self._max_length_resp_file_name:
+                    self._max_length_resp_file_name = len(pha_info['rsp'].rsp_filename)
+
+            else:
+
+                # This will be reached in the case that a response was generated from a plugin
+                # e.g. if we want to use weighted DRMs from GBM. We do not handle this just yet.
+
+                NotImplementedError("In the future this indicates that we need to generate an RSP file.")
+
+            self._rate[key].append(pha_info[key].rates.tolist())
+
+            self._backscal[key].append(pha_info[key].scale_factor)
+
+            if not pha_info[key].is_poisson():
+
+                self._is_poisson[key] = pha_info[key].is_poisson()
+
+                self._stat_err[key].append(pha_info[key].rate_errors.tolist())
+
+            # If there is systematic error, we add it
+            # otherwise create an array of zeros as XSPEC
+            # simply adds systematic in quadrature to statistical
+            # error.
+
+            if pha_info[key].sys_errors.tolist() is not None:  # It returns an array which does not work!
+
+                self._sys_err[key].append(pha_info[key].sys_errors.tolist())
+
+            else:
+
+                self._sys_err[key].append(np.zeros_like(pha_info[key].rates, dtype=np.float32).tolist())
+
+            self._exposure[key].append(pha_info[key].exposure)
+            self._quality[key].append(ogip.ogip_quality.tolist())
+            self._grouping[key].append(ogip.ogip_grouping.tolist())
+            self._channel[key].append(np.arange(pha_info[key].n_channels, dtype=np.int8) + first_channel)
+
+            if ogip.tstart is not None:
+
+                self._tstart[key].append(ogip.tstart)
+
+                if ogip.tstop is not None:
+
+                    self._tstop[key].append(ogip.tstop)
+
+                else:
+
+                    RuntimeError('OGIP TSTART is a number but TSTOP is None. This is a bug.')
+
+            # We will assume that the exposure is the true DT
+            # and assign starts and stops accordingly. This means
+            # we most likely are dealing with a simulation.
+            else:
+
+                self._tstart[key].append(self._pseudo_time)
+
+                self._pseudo_time += pha_info[key].exposure
+
+                self._tstop[key].append(self._pseudo_time)
+
+    def _write_phaII(self, overwrite):
+
+        trigTime = None
+        clobber = False
+
+        # Assuming background and pha files have the same
```

Comment:
if I understand what this means, I think this should be enforced with an "assert" clause.